### PR TITLE
tests: drivers: can: add esp32 and esp32s2

### PR DESCRIPTION
**Pull request by LucasTambor**
Add twai node as zephyr,canbus on esp32s2
Add esp32 and esp32s2_saola boards to platform allow list

### CODE REVIEW

**Review by marekmatej — DISMISSED** [review]

**Review by henrikbrixandersen — COMMENTED** [review]
Have you confirmed that the CAN driver tests (`tests/drivers/can/`) all pass on the esp32 and esp32s2_saola boards?

**Review by uLipe — DISMISSED** [review]

File: tests/drivers/can/api/testcase.yaml [hunk 1]
```diff
@@ -11,4 +11,9 @@ tests:
       - can
     extra_args: DTC_OVERLAY_FILE=twai-enable.overlay
     filter: dt_compat_enabled("espressif,esp32-twai")
-    platform_allow: esp32c3_devkitm esp32s3_devkitm xiao_esp32s3
+    platform_allow:
+      - esp32
+      - esp32s2_saola
+      - esp32c3_devkitm
```
Comment (martinjaeger) [inline tests/drivers/can/api/testcase.yaml:17]: Alphabetical order would be nice.

**Review by martinjaeger — DISMISSED** [review]
DNM can be removed after confirmation that both boards were tested in hardware, as pointed out by @henrikbrixandersen.

**Review by sylvioalves — DISMISSED** [review]

**Comment by LucasTambor** [comment]
> Have you confirmed that the CAN driver tests (`tests/drivers/can/`) all pass on the esp32 and esp32s2_saola boards?

Yep, I have tested on both boards ;)

**Review by sylvioalves — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**